mimic/model/clb_objects.py

```diff
@@ -38,6 +42,28 @@ def __init__(self):
         self.lbs = {}
         self.meta = {}
 
+    def __len__(self):
```

Comment:
I generally avoid making objects special unless they're going to be "all the way" special, and documented as such.  For example, this change makes `RegionalCLBCollection` somewhat list-like (as you discovered, it is now accidentally iterable) but also somewhat mapping-like.  However, although it is mapping-like it is missing big pieces of the mapping interface: `.__setitem__(...)`, `.clear()`, `.copy()`, `.get(...)`, `.has_key()`, `.keys()`, and so on and so on.  It also gets rid of the symmetry of `RegionalCLBCollection` with other similar types in e.g. the Nova mock which are not partial-mapping-like.

The reason that this not-quite-a-dict-like thing is important is that it becomes a point that maintainers can easily be tripped up on.  When looking at a variable in python code, dynamic typing means you don't have to know which type it necessarily *is*, but in order to be able to make substantive changes you need to know what kind of type it is supposed to be *like*.  If you have tedious method names like `load_balancer_with_names` or `load_balancer_exists`, then it's very easy to go from a caller to a `grep` command to the implementation.

Then there's the issue of the fact that there are two attributes here: `lbs` and `meta`, but we are only forwarding mappish operations to `lbs`.

Finally, a convenient syntax like this should be reserved for a highly polished type with really clear `repr`s and so on so that any clarity lost to using this convenience API is made up for in smooth runtime error reporting and formatting in debug messages.  That makes this a poor *time* to start adding special methods, because the returned values here are going to be simple dictionaries, and if you end up staring at one in a debugger or a stack dump, you won't necessarily know where it came from unless you happen to know what format our idiom of a "load balancer" dictionary happens to look like at the moment.

*Finally* finally there's also the issue that not all of these methods are even used; some of them are responsible for the coverage decrease currently failing the builders.